modules/apps/headless/headless-admin-workflow/headless-admin-workflow-test/src/testIntegration/java/com/liferay/headless/admin/workflow/resource/v1_0/test/WorkflowLogResourceTest.java

```diff
@@ -15,14 +15,388 @@
+				new WorkflowLog() {
+					{
+						commentLog = _language.get(
+							_getResourceBundle(), "assigned-initial-task");
+						description = _language.format(
+							_getResourceBundle(),
+							"task-initially-assigned-to-the-x-role",
+							_siteContentReviewerRole.getTitle(
+								LocaleUtil.getDefault()),
+							false);
+						state = "review";
+						type = Type.TASK_ASSIGN;
+						workflowTaskId = workflowTask.getId();
+					}
+				}),
+			(List<WorkflowLog>)page.getItems());
+
+		assertValid(page);
+	}
+
+	@Override
+	@Test
+	public void testGetWorkflowTaskWorkflowLogsPageWithPagination()
+		throws Exception {
+
+		WorkflowTask workflowTask = WorkflowTaskTestUtil.getWorkflowTask(
+			_workflowInstance.getId());
+
+		_workflowTaskManager.assignWorkflowTaskToUser(
+			TestPropsValues.getCompanyId(), TestPropsValues.getUserId(),
+			workflowTask.getId(), TestPropsValues.getUserId(), StringPool.BLANK,
+			null, null);
+
+		Page<WorkflowLog> page1 =
+			workflowLogResource.getWorkflowTaskWorkflowLogsPage(
+				workflowTask.getId(),
+				new String[] {WorkflowLog.Type.TASK_ASSIGN.getValue()},
+				Pagination.of(1, 1));
+
+		List<WorkflowLog> workflowLogs1 = (List<WorkflowLog>)page1.getItems();
+
+		Assert.assertEquals(workflowLogs1.toString(), 1, workflowLogs1.size());
+
+		assertEquals(
+			Arrays.asList(
+				new WorkflowLog() {
+					{
+						commentLog = StringPool.BLANK;
+						description = _language.format(
+							_getResourceBundle(),
+							"x-assigned-the-task-to-himself",
+							new Object[] {
+								_portal.getUserName(
+									TestPropsValues.getUserId(),
+									StringPool.BLANK),
+								_portal.getUserName(
+									TestPropsValues.getUserId(),
+									StringPool.BLANK)
+							},
+							false);
+						state = "review";
+						type = Type.TASK_ASSIGN;
+						workflowTaskId = workflowTask.getId();
+					}
+				}),
+			workflowLogs1);
+
+		Page<WorkflowLog> page2 =
+			workflowLogResource.getWorkflowTaskWorkflowLogsPage(
+				workflowTask.getId(),
+				new String[] {WorkflowLog.Type.TASK_ASSIGN.getValue()},
+				Pagination.of(2, 1));
+
+		Assert.assertEquals(2, page2.getTotalCount());
+
+		List<WorkflowLog> workflowLogs2 = (List<WorkflowLog>)page2.getItems();
+
+		Assert.assertEquals(workflowLogs2.toString(), 1, workflowLogs2.size());
+
+		assertEquals(
+			Arrays.asList(
+				new WorkflowLog() {
+					{
+						commentLog = _language.get(
+							_getResourceBundle(), "assigned-initial-task");
+						description = _language.format(
+							_getResourceBundle(),
+							"task-initially-assigned-to-the-x-role",
+							_siteContentReviewerRole.getTitle(
+								LocaleUtil.getDefault()),
+							false);
+						state = "review";
+						type = Type.TASK_ASSIGN;
+						workflowTaskId = workflowTask.getId();
+					}
+				}),
+			workflowLogs2);
+	}
+
+	@Override
+	protected String[] getAdditionalAssertFieldNames() {
+		return new String[] {
+			"commentLog", "description", "state", "workflowTaskId"
```

Comment:
Hi, @seltong

I guess the `"type"` an interesting field to compare in equals method, what do you think?